dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/WebContainer.java

```diff
@@ -1378,12 +1378,11 @@ public static void addHttpServletResponseListener(IHttpServletResponseListener l
 
     public static void notifyHttpServletResponseListenersPreHeaderCommit(HttpServletRequest request, HttpServletResponse response) {
         // need to notify listeners registered in the ServletRequestAttributeListener array
-        if (!httpResponseListeners.isEmpty()) {
-            Iterator i = httpResponseListeners.iterator();
-
-            while (i.hasNext()) {
+        int listenerSize = httpResponseListeners.size();
+        if (listenerSize != 0) {
+            for (int i = 0; i < listenerSize; ++i) {
```

Comment:
just for my information, is loop more effective than iteration?

Reply:
This is avoiding creating an iterator.